flink/v1.14/flink/src/main/java/com/netease/arctic/flink/read/hybrid/enumerator/ContinuousSplitPlannerImpl.java

```diff
@@ -72,16 +73,61 @@ public ContinuousEnumerationResult planSplits(ArcticEnumeratorOffset lastOffset)
     }
   }
 
+  @Override
+  public ContinuousEnumerationResult planSplits(ArcticEnumeratorOffset lastOffset, List<Expression> filters) {
+    if (table == null) {
+      table = loadArcticTable(loader).asKeyedTable();
+    }
+    table.refresh();
+    if (lastOffset != null) {
+      return discoverIncrementalSplits(lastOffset, filters);
+    } else {
+      return discoverInitialSplits(filters);
+    }
+  }
+
   private ContinuousEnumerationResult discoverIncrementalSplits(ArcticEnumeratorOffset lastPosition) {
     long fromChangeSnapshotId = lastPosition.changeSnapshotId();
     Snapshot changeSnapshot = table.changeTable().currentSnapshot();
     if (changeSnapshot != null && changeSnapshot.snapshotId() != fromChangeSnapshotId) {
+      Long fromSequence = null;
+      if (fromChangeSnapshotId != Long.MIN_VALUE) {
+        Snapshot snapshot = table.changeTable().snapshot(fromChangeSnapshotId);
+        fromSequence = snapshot.sequenceNumber();
+      }
+
       long snapshotId = changeSnapshot.snapshotId();
       TableEntriesScan entriesScan = TableEntriesScan.builder(table.changeTable())
           .useSnapshot(snapshotId)
           .includeFileContent(FileContent.DATA)
+          .fromSequence(fromSequence)
           .build();
 
+      List<ArcticSplit> arcticChangeSplit =
+          planChangeTable(entriesScan, fromSequence, table.changeTable().spec(), splitCount);
+      return new ContinuousEnumerationResult(
+          arcticChangeSplit,
+          lastPosition,
+          ArcticEnumeratorOffset.of(snapshotId, null));
+    }
+    return ContinuousEnumerationResult.EMPTY;
+  }
+
+  protected ContinuousEnumerationResult discoverIncrementalSplits(
```

Comment:
These two `discoverIncrementalSplits` have so much identical code, perhaps we can merge them.